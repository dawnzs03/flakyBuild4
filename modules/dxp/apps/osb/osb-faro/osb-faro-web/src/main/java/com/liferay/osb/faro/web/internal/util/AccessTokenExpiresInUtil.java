/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

/**
 * @author Ivica Cardic
 */
public class AccessTokenExpiresInUtil {

	public static long getExpiresIn() {
		return _expiresIn;
	}

	public static void removeExpiresIn() {
		_expiresIn = 0;
	}

	public static void setExpiresIn(long expiresIn) {
		_expiresIn = expiresIn;
	}

	private static long _expiresIn;

}