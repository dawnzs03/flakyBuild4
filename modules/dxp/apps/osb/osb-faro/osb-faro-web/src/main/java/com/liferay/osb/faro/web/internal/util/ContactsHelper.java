/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.engine.client.ContactsEngineClient;
import com.liferay.osb.faro.engine.client.constants.FieldMappingConstants;
import com.liferay.osb.faro.engine.client.model.Individual;
import com.liferay.osb.faro.engine.client.model.IndividualSegment;
import com.liferay.osb.faro.model.FaroProject;
import com.liferay.osb.faro.web.internal.constants.FaroConstants;
import com.liferay.osb.faro.web.internal.model.display.contacts.IndividualDisplay;
import com.liferay.osb.faro.web.internal.model.display.contacts.IndividualSegmentDisplay;
import com.liferay.osb.faro.web.internal.model.display.main.FaroEntityDisplay;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Shinn Lok
 */
@Component(service = ContactsHelper.class)
public class ContactsHelper {

	public FaroEntityDisplay getContactsEntityDisplay(
		FaroProject faroProject, String contactsEntityId, int type) {

		if (type == FaroConstants.TYPE_INDIVIDUAL) {
			Individual individual = _contactsEngineClient.getIndividual(
				faroProject, contactsEntityId, null);

			return new IndividualDisplay(individual);
		}

		IndividualSegment individualSegment =
			_contactsEngineClient.getIndividualSegment(
				faroProject, contactsEntityId, false);

		return new IndividualSegmentDisplay(individualSegment);
	}

	public String getOwnerType(int type) {
		if (type == FaroConstants.TYPE_ACCOUNT) {
			return FieldMappingConstants.OWNER_TYPE_ACCOUNT;
		}
		else if (type == FaroConstants.TYPE_INDIVIDUAL) {
			return FieldMappingConstants.OWNER_TYPE_INDIVIDUAL;
		}

		return FieldMappingConstants.OWNER_TYPE_INDIVIDUAL_SEGMENT;
	}

	@Reference
	private ContactsEngineClient _contactsEngineClient;

}