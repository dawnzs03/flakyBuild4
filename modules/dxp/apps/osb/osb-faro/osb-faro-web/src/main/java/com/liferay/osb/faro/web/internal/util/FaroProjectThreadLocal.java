/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.model.FaroProject;
import com.liferay.petra.lang.CentralizedThreadLocal;

/**
 * @author Marcellus Tavares
 */
public class FaroProjectThreadLocal {

	public static FaroProject getFaroProject() {
		return _faroProject.get();
	}

	public static void removeFaroProject() {
		_faroProject.remove();
	}

	public static void setFaroProject(FaroProject faroProject) {
		_faroProject.set(faroProject);
	}

	private static final ThreadLocal<FaroProject> _faroProject =
		new CentralizedThreadLocal<>(
			FaroProjectThreadLocal.class + "._faroProject");

}