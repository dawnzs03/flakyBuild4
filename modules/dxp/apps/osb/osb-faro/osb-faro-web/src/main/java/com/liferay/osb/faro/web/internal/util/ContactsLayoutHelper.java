/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.contacts.model.ContactsCardTemplate;
import com.liferay.osb.faro.contacts.service.ContactsCardTemplateLocalService;
import com.liferay.osb.faro.web.internal.card.template.ContactsCardTemplateManagerHelper;
import com.liferay.osb.faro.web.internal.model.display.contacts.card.template.ContactsLayoutTemplateSettingDisplay;
import com.liferay.petra.string.StringBundler;
import com.liferay.petra.string.StringPool;
import com.liferay.portal.kernel.model.UserConstants;

import java.util.ArrayList;
import java.util.List;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Matthew Kong
 */
@Component(service = ContactsLayoutHelper.class)
public class ContactsLayoutHelper {

	public String addContactsCardTemplateIds(long groupId, int type)
		throws Exception {

		List<List<ContactsLayoutTemplateSettingDisplay>>
			contactsLayoutTemplateSettingDisplaysList = new ArrayList<>();

		for (int[] contactsCardTemplateTypes :
				_contactsCardTemplateManagerUtil.
					getDefaultContactsCardTemplateTypes(type)) {

			List<ContactsLayoutTemplateSettingDisplay>
				contactsLayoutTemplateSettingDisplays = new ArrayList<>();

			for (int contactsCardTemplateType : contactsCardTemplateTypes) {
				ContactsCardTemplate contactsCardTemplate =
					_contactsCardTemplateLocalService.addContactsCardTemplate(
						groupId, UserConstants.USER_ID_DEFAULT,
						_contactsCardTemplateManagerUtil.getDefaultName(
							contactsCardTemplateType),
						_contactsCardTemplateManagerUtil.getDefaultSettings(
							contactsCardTemplateType),
						contactsCardTemplateType);

				contactsLayoutTemplateSettingDisplays.add(
					new ContactsLayoutTemplateSettingDisplay(
						contactsCardTemplate.getContactsCardTemplateId(), 0));
			}

			contactsLayoutTemplateSettingDisplaysList.add(
				contactsLayoutTemplateSettingDisplays);
		}

		return JSONUtil.writeValueAsString(
			contactsLayoutTemplateSettingDisplaysList);
	}

	public String addHeaderContactsCardTemplateIds(long groupId, int type) {
		int[] contactsCardTemplateTypes =
			_contactsCardTemplateManagerUtil.
				getDefaultHeaderContactsCardTemplateTypes(type);

		if (contactsCardTemplateTypes.length == 0) {
			return StringPool.BLANK;
		}

		StringBundler sb = new StringBundler(
			contactsCardTemplateTypes.length * 2);

		for (int contactsCardTemplateType : contactsCardTemplateTypes) {
			ContactsCardTemplate contactsCardTemplate =
				_contactsCardTemplateLocalService.addContactsCardTemplate(
					groupId, UserConstants.USER_ID_DEFAULT,
					_contactsCardTemplateManagerUtil.getDefaultName(
						contactsCardTemplateType),
					_contactsCardTemplateManagerUtil.getDefaultSettings(
						contactsCardTemplateType),
					contactsCardTemplateType);

			sb.append(contactsCardTemplate.getContactsCardTemplateId());

			sb.append(StringPool.COMMA);
		}

		sb.setIndex(sb.index() - 1);

		return sb.toString();
	}

	@Reference
	private ContactsCardTemplateLocalService _contactsCardTemplateLocalService;

	@Reference
	private ContactsCardTemplateManagerHelper _contactsCardTemplateManagerUtil;

}