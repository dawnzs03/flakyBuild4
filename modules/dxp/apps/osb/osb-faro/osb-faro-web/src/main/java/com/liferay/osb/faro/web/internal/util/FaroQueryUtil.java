/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.petra.string.StringPool;
import com.liferay.portal.kernel.util.StringUtil;
import com.liferay.portal.kernel.util.Validator;

import java.util.stream.Collectors;
import java.util.stream.IntStream;

/**
 * @author Gabriel Ibson
 */
public class FaroQueryUtil {

	public static String sanitizeQuery(String query) {
		if (Validator.isBlank(query)) {
			return query;
		}

		if (StringUtil.equalsIgnoreCase(query, StringPool.NULL)) {
			return "\\null";
		}

		IntStream intStream = query.codePoints();

		return intStream.mapToObj(
			c -> (char)c
		).map(
			c -> {
				if (_CHARACTERS_TO_BE_ESCAPED.indexOf(c) >= 0) {
					return "\\" + c;
				}

				return String.valueOf(c);
			}
		).collect(
			Collectors.joining()
		);
	}

	private static final String _CHARACTERS_TO_BE_ESCAPED =
		"+-=&&||><!(){}[]^\"~*?:\\/%";

}