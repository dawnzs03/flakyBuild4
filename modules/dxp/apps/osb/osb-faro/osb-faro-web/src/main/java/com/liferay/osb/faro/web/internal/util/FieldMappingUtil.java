/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.engine.client.ContactsEngineClient;
import com.liferay.osb.faro.engine.client.model.FieldMapping;
import com.liferay.osb.faro.engine.client.model.FieldMappingMap;
import com.liferay.osb.faro.engine.client.model.Results;
import com.liferay.osb.faro.model.FaroProject;
import com.liferay.petra.function.transform.TransformUtil;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

/**
 * @author Matthew Kong
 */
public class FieldMappingUtil {

	public static List<FieldMappingMap> getNewFieldMappingMaps(
		ContactsEngineClient contactsEngineClient, FaroProject faroProject,
		String context, List<FieldMappingMap> fieldMappingMaps) {

		List<FieldMappingMap> newFieldMappingMaps = new ArrayList<>();

		Results<FieldMapping> results = contactsEngineClient.getFieldMappings(
			faroProject, context,
			TransformUtil.transform(fieldMappingMaps, FieldMappingMap::getName),
			1, 10000, null);

		Set<String> currentFieldNames = new HashSet<>();

		for (FieldMapping fieldMapping : results.getItems()) {
			currentFieldNames.add(fieldMapping.getFieldName());
		}

		Set<String> newFieldMappingNames = new HashSet<>();

		for (FieldMappingMap fieldMappingMap : fieldMappingMaps) {
			String name = fieldMappingMap.getName();

			if (currentFieldNames.contains(name) ||
				newFieldMappingNames.contains(name)) {

				continue;
			}

			newFieldMappingNames.add(name);

			newFieldMappingMaps.add(fieldMappingMap);
		}

		return newFieldMappingMaps;
	}

}