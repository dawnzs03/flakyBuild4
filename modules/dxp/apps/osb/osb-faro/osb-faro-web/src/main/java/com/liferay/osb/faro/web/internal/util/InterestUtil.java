/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.engine.client.ContactsEngineClient;
import com.liferay.osb.faro.engine.client.model.Interest;
import com.liferay.osb.faro.engine.client.model.Rels;
import com.liferay.osb.faro.engine.client.model.Results;
import com.liferay.osb.faro.engine.client.util.OrderByField;
import com.liferay.osb.faro.model.FaroProject;
import com.liferay.osb.faro.web.internal.model.display.FaroResultsDisplay;
import com.liferay.osb.faro.web.internal.model.display.contacts.InterestDisplay;

import java.util.List;
import java.util.function.Function;

/**
 * @author Shinn Lok
 */
public class InterestUtil {

	@SuppressWarnings("unchecked")
	public static FaroResultsDisplay getInterests(
		FaroProject faroProject, String channelId, String contactsEntityId,
		String query, int cur, int delta, List<OrderByField> orderByFields,
		ContactsEngineClient contactsEngineClient) {

		Results<Interest> results = contactsEngineClient.getInterests(
			faroProject, channelId, contactsEntityId, null, null, query,
			Rels.Interests.PAGES_VISITED, cur, delta, orderByFields);

		Function<Interest, InterestDisplay> function = InterestDisplay::new;

		return new FaroResultsDisplay(results, function);
	}

}