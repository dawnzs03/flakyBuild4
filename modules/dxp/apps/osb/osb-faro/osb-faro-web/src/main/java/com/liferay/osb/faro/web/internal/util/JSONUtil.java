/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.PropertyAccessor;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.JavaType;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.cfg.MapperConfig;
import com.fasterxml.jackson.databind.introspect.AnnotatedField;

import com.liferay.osb.faro.engine.client.model.Credentials;
import com.liferay.osb.faro.engine.client.model.Interest;
import com.liferay.osb.faro.engine.client.model.PageVisited;
import com.liferay.osb.faro.engine.client.model.credentials.OAuth1Credentials;
import com.liferay.osb.faro.engine.client.model.credentials.OAuth2Credentials;
import com.liferay.osb.faro.engine.client.model.credentials.TokenCredentials;
import com.liferay.osb.faro.web.internal.model.display.contacts.mixin.BaseMixin;
import com.liferay.osb.faro.web.internal.model.display.contacts.mixin.CredentialsMixin;
import com.liferay.osb.faro.web.internal.model.display.contacts.mixin.OAuth1CredentialsMixin;
import com.liferay.osb.faro.web.internal.model.display.contacts.mixin.OAuth2CredentialsMixin;
import com.liferay.osb.faro.web.internal.model.display.contacts.mixin.PageVisitMixin;
import com.liferay.osb.faro.web.internal.model.display.contacts.mixin.TokenCredentialsMixin;
import com.liferay.petra.string.StringPool;
import com.liferay.portal.kernel.util.StringUtil;

import java.lang.reflect.Type;

/**
 * @author Matthew Kong
 */
public class JSONUtil {

	public static <T> T convertValue(Object object, Class<T> clazz) {
		return _objectMapper.convertValue(object, clazz);
	}

	public static <T> T convertValue(
		Object object, TypeReference<T> typeReference) {

		return _objectMapper.convertValue(object, typeReference);
	}

	public static ObjectMapper getObjectMapper() {
		return _objectMapper;
	}

	public static <T> T readContainedTypeValue(String content, Type type)
		throws Exception {

		JavaType javaType = _objectMapper.constructType(type);

		javaType = javaType.containedType(0);

		return _objectMapper.readValue(content, javaType);
	}

	public static <T> T readValue(String content, Class<T> valueType)
		throws Exception {

		return _objectMapper.readValue(content, valueType);
	}

	public static <T> T readValue(
			String content, TypeReference<T> typeReference)
		throws Exception {

		return _objectMapper.readValue(content, typeReference);
	}

	public static String writeValueAsString(Object value) throws Exception {
		return _objectMapper.writeValueAsString(value);
	}

	private static final ObjectMapper _objectMapper = new ObjectMapper() {
		{
			addMixIn(Credentials.class, CredentialsMixin.class);
			addMixIn(Interest.class, BaseMixin.class);
			addMixIn(OAuth1Credentials.class, OAuth1CredentialsMixin.class);
			addMixIn(OAuth2Credentials.class, OAuth2CredentialsMixin.class);
			addMixIn(PageVisited.class, PageVisitMixin.class);
			addMixIn(TokenCredentials.class, TokenCredentialsMixin.class);
			configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);

			setPropertyNamingStrategy(
				new PropertyNamingStrategy() {

					@Override
					public String nameForField(
						MapperConfig<?> config, AnnotatedField field,
						String defaultName) {

						if (defaultName.startsWith(StringPool.UNDERLINE)) {
							defaultName = defaultName.substring(1);
						}

						defaultName = StringUtil.removeSubstring(
							defaultName, "Display");

						return super.nameForField(config, field, defaultName);
					}

				});

			setVisibility(
				PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
		}
	};

}