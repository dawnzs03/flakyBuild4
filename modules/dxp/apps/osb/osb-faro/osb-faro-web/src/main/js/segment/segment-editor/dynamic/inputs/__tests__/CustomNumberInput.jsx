import CustomNumberInput from '../CustomNumberInput';
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import {fromJS} from 'immutable';
import {Property} from 'shared/util/records';

jest.unmock('react-dom');

describe('CustomNumberInput', () => {
	afterEach(cleanup);

	it('should render', () => {
		const {container, getAllByText, getByText} = render(
			<CustomNumberInput
				property={new Property()}
				touched={false}
				valid={false}
				value={fromJS({
					criterionGroup: {items: [{operatorName: 'eq', value: ''}]}
				})}
			/>
		);
		fireEvent.click(getByText('is equal to'));

		expect(getAllByText('is equal to')[1]).toBeTruthy();
		expect(getByText('greater than')).toBeTruthy();
		expect(getByText('less than')).toBeTruthy();
		expect(getByText('is not equal to')).toBeTruthy();
		expect(getByText('is known')).toBeTruthy();
		expect(getByText('is unknown')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('should render with data', () => {
		const {container, getAllByText, getByText} = render(
			<CustomNumberInput
				displayValue='Revenue'
				operatorRenderer={() => <div>{'operator'}</div>}
				property={new Property()}
				touched={false}
				valid
				value={fromJS({
					criterionGroup: {
						items: [{operatorName: 'eq', value: '123'}]
					}
				})}
			/>
		);
		fireEvent.click(getByText('is equal to'));

		expect(getAllByText('is equal to')[1]).toBeTruthy();
		expect(getByText('greater than')).toBeTruthy();
		expect(getByText('less than')).toBeTruthy();
		expect(getByText('is not equal to')).toBeTruthy();
		expect(getByText('is known')).toBeTruthy();
		expect(getByText('is unknown')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('should render w/o value input when value is null', () => {
		const {queryByTestId} = render(
			<CustomNumberInput
				displayValue='Revenue'
				operatorRenderer={() => <div>{'operator'}</div>}
				property={new Property()}
				touched={false}
				valid
				value={fromJS({
					criterionGroup: {
						items: [{operatorName: 'eq', value: null}]
					}
				})}
			/>
		);

		expect(queryByTestId('number-input')).toBeNull();
	});

	it('should render w/ has-error when touched and not valid', () => {
		const {container} = render(
			<CustomNumberInput
				displayValue='Revenue'
				operatorRenderer={() => <div>{'operator'}</div>}
				property={new Property()}
				touched
				valid={false}
				value={fromJS({
					criterionGroup: {items: [{operatorName: 'eq', value: ''}]}
				})}
			/>
		);

		expect(container.querySelector('.has-error')).toBeTruthy();
	});
});
