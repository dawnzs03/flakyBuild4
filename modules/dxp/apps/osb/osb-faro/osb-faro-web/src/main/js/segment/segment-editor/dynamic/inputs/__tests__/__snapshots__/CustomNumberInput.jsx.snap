// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomNumberInput should render 1`] = `
<div>
  <div
    class="criteria-statement"
  >
    <div
      class="form-group form-group-autofit"
    >
      <div
        aria-hidden="true"
        class="form-group-item entity-name form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      />
      <div
        aria-hidden="true"
        class="form-group-item display-value form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      />
      <div
        class="form-group-item form-group-item-shrink"
      >
        <button
          aria-activedescendant="eq"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="clay-id-2"
          class="form-control form-control-select form-control-select-secondary show"
          role="combobox"
          style=""
          tabindex="0"
          type="button"
        >
          is equal to
        </button>
      </div>
      <div
        aria-hidden="true"
        class="form-group-item form-group-item-shrink"
        data-aria-hidden="true"
      >
        <input
          class="input-root form-control"
          data-testid="number-input"
          type="number"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`CustomNumberInput should render with data 1`] = `
<div>
  <div
    class="criteria-statement"
  >
    <div
      class="form-group form-group-autofit"
    >
      <div
        aria-hidden="true"
        class="form-group-item entity-name form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      />
      <div
        aria-hidden="true"
        class="form-group-item display-value form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      >
        Revenue
      </div>
      <div
        class="form-group-item form-group-item-shrink"
      >
        <button
          aria-activedescendant="eq"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="clay-id-4"
          class="form-control form-control-select form-control-select-secondary show"
          role="combobox"
          style=""
          tabindex="0"
          type="button"
        >
          is equal to
        </button>
      </div>
      <div
        aria-hidden="true"
        class="form-group-item form-group-item-shrink"
        data-aria-hidden="true"
      >
        <input
          class="input-root form-control"
          data-testid="number-input"
          type="number"
          value="123"
        />
      </div>
    </div>
  </div>
</div>
`;
